### PR TITLE
Decaffeinate `spec/ci-spec`

### PR DESCRIPTION
**Pull request by GuilleW**
Since I work on many files, I edit `package.json` to add useful shortcut to avoid misspelt filename.

```json
    "test-decaff": "npx coffee --compile --output spec spec/${npm_config_name}.coffee",
    "test-coffee": "jasmine-focused --captureExceptions --coffee spec/${npm_config_name}.coffee",
    "test-js": "jasmine-focused --captureExceptions spec/${npm_config_name}.js"
```

## Steps to reproduce decaffeination ([guidelines](https://github.com/pulsar-edit/.github/blob/main/project-birth/CONTRIBUTING-DURING-START.md#decaffeination)):

- [x] Check package.json for coffee-script version
  - version: 1.12.7
- [x] Decaffeination with official Coffee program
  - `npx coffee --compile --output spec spec/ci-spec.coffee`
  - (or my shortcut:  `./bin/npm run test-decaff --name=ci-spec`)
- [x] Quality of Decaffeination
  - [x] Replace `var` with `const` and `let` if it's possible.
  - [x] Replace Nested [Ternary Operators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) with if statements.
  - [x] Remove `return` when it's not needed.
  - [x] Remove Result Arrays when they are not needed.
  - [x] Replace for loops with [for-of loops](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of).
  - [x] Use [Arrow Functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) to write functions in a single line when it's possible.
  - [x] Remove unneeded ref variables.
  - [x] Remove [do blocks](https://devdocs.io/coffeescript~1/index#loops) and use let or const instead.
  - [x] Tests and assertions should be the same and success like in `coffee` file
  
## Test from coffee and JS should output same successful result :

From coffee
```sh
$ jasmine-focused --captureExceptions --coffee spec/ci-spec.coffee
// or my shortcut $ `./bin/npm run test-coffee --name=ci-spec`
> 4 tests, 8 assertions, 0 failures, 0 skipped
```
to js
```sh
$ jasmine-focused --captureExceptions spec/ci-spec.js
// or my shortcut $ `./bin/npm run test-js --name=ci-spec`
> 4 tests, 8 assertions, 0 failures, 0 skipped
```

### CODE REVIEW

File: spec/ci-spec.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      process.env.ATOM_PACKAGES_URL = 'http://localhost:3000/packages'
+      process.env.ATOM_ELECTRON_VERSION = 'v10.20.1'
+      process.env.npm_config_registry = 'http://localhost:3000/'
+      live = true
+    })
+    waitsFor(() => live)
+  })
+
+  afterEach(() => {
+    var done = false
```
Comment (confused-Techie) [inline spec/ci-spec.js:66]:
```suggestion
    let done = false
```

File: spec/ci-spec.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      response.sendFile(path.join(__dirname, 'fixtures', 'test-module-with-dependencies-1.1.0.tgz'))
+    })
+    app.get('/tarball/test-module-1.1.0.tgz', (request, response) => {
+      response.sendFile(path.join(__dirname, 'fixtures', 'test-module-1.1.0.tgz'))
+    })
+    app.get('/tarball/native-module-1.0.0.tgz', (request, response) => {
+      response.sendFile(path.join(__dirname, 'fixtures', 'native-module-1.0.0.tgz'))
+    })
+    server = http.createServer(app)
+    var live = false
```
Comment (confused-Techie) [inline spec/ci-spec.js:54]:
```suggestion
    let live = false
```

Comment (confused-Techie) [inline spec/ci-spec.js:54]:
```suggestion
    let live = false
```

File: spec/ci-spec.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+const fs = require('fs')
+const http = require('http')
+const temp = require('temp')
+const express = require('express')
+const wrench = require('wrench')
+const CSON = require('season')
+const apm = require('../lib/apm-cli')
+
+describe('apm ci', () => {
+  var server
```
Comment (confused-Techie) [inline spec/ci-spec.js:11]:
```suggestion
  let server
```

**Review by confused-Techie — CHANGES_REQUESTED** [review]
Overall this PR looks great, just some simple changes requested, otherwise lets get it merged!

**Comment by 2colours** [comment]
Supposedly superseeded by https://github.com/pulsar-edit/ppm/pull/65

**Comment by confused-Techie** [comment]
Superseded by #65 

Thanks again @GuilleW for your fantastic work here, sorry it has taken some time to get it merged in some form. Appreciate your efforts!

**Pull request closed without merging**